inc/Engine/Optimization/RUCSS/Admin/Subscriber.php

```diff
@@ -205,6 +209,34 @@ public function clean_used_css_and_cache( $value, $old_value ) {
 			$this->database->truncate_used_css_table();
 			// Clear all caching files.
 			rocket_clean_domain();
+			$this->used_css->delete_all_used_css_files();
+		}
+	}
+
+	/**
+	 * Clean used_css directory with CPCSS option disabled.
+	 *
+	 * @since 3.9
+	 *
+	 * @param array $old_value Options old value.
+	 * @param array $value Options new value.
+	 */
+	public function clean_used_css_directory_cpcss_disabled( array $old_value, array $value ) {
+		if ( ! current_user_can( 'rocket_manage_options' )
+			||
+			! $this->settings->is_enabled()
+		) {
+			return;
+		}
+
+		if (
+			isset( $value['async_css'], $old_value['async_css'] )
+			&&
+			$value['async_css'] !== $old_value['async_css']
+			&&
+			! (bool) $value['async_css']
+		) {
+			$this->used_css->delete_all_used_css_files();
```

Comment:
I think in here we need to clean also Rocket Cache?

Or is handled in Critical CSS code when CPCSS is disabled?

Reply:
cache is removed with saving options, with cpcss and any other option